examples/syntax.nit

```diff
@@ -0,0 +1,182 @@
+#! /usr/bin/env nitc
+
+# This file is part of NIT ( http://www.nitlanguage.org ).
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Test case for syntax highlighters.
+#
+# Also test spell-checking and comment recognition.
+#
+# SEE: [Nit](http://nitlanguage.org)
+# XXX: Dummy task
+module syntax is
+	new_annotation new_annotation
+
+	conditional
+
+	deprecated
+	fixed
+	lazy
+	noinit
+	readonly
+	writable
+	optional
+	autoinit
+	noautoinit
+	lateinit
+	nosuper
+	old_style_init
+	abstract
+	intern
+	extern
+	no_warning("fictive-module")
+	generated
+
+	pkgconfig("a libray from outer space")
```

Comment:
I didn't see the `example` directory was used. I think there is two distinct goals:
1. to show the various syntax constructions of Nit to a new user with a good signal/noise ratio and without frightening him.
2. to express comprehensive pieces of syntax constructions (with rare cases and pitfalls) to test code-level tools like highlighters.

Since it seems to me that the goal of the file is 2. I suggest that you move the file to tests/
I also suggests that you add a commit that runs nitlight and nitpretty on this file.

Even if the file is a demo for now and require visual inspection on the result. One can imagine adding some CI feature latter, where adding a new syntax construction will require an update on the file and where an update on the file will require to update&check the various language definitions.